functions/Backup-DbaDatabase.ps1

```diff
@@ -453,8 +453,17 @@ function Backup-DbaDatabase {
 
             if ($CompressBackup) {
                 if ($db.EncryptionEnabled) {
-                    Write-Message -Level Warning -Message "$dbName is enabled for encryption, will not compress"
-                    $backup.CompressionOption = 2
+                    $minVerForTDECompression = [version]'13.0.4446.0' #SQL Server 2016 CU 4
+                    $flagTDESQLVersion = $minVerForTDECompression -le $Server.version
+                    $flagTestBoundMaxTransferSize = Test-Bound 'MaxTransferSize'
+                    $flagCorrectMaxTransferSize = $flagTestBoundMaxTransferSize -and ($MaxTransferSize -gt 64kb)
+                    if ($flagTDESQLVersion -and $flagTestBoundMaxTransferSize -and $flagCorrectMaxTransferSize) {
+                        Write-Message -Level Verbose "$dbName is enabled for encryption but will compress"
+                        $backup.CompressionOption = 1
+                    } else {
+                        Write-Message -Level Output -Message "$dbName is enabled for encryption, will not compress"
```

Comment:
Why -Level Output? On the old version it was -Level Warning.

Reply:
In the previous version, someone required a compressed backup and the module disable compression in an authorative way.
In this version compression is performed as required so I suppose that there is no reason to issue a warning,

Reply:
No, I'm talking about these lines:
```
                    } else {
                        Write-Message -Level Output -Message "$dbName is enabled for encryption, will not compress"
                        $backup.CompressionOption = 2
                    }
```

In the old version this is this code:
```
               if ($db.EncryptionEnabled) {
                    Write-Message -Level Warning -Message "$dbName is enabled for encryption, will not compress"
                    $backup.CompressionOption = 2
```

So same result ($backup.CompressionOption = 2), but different message.

Reply:
I'll correct